### ISSUE TITLE
How to change background color for a time range?

### ISSUE THREAD

**Issue by nizarbathery** [antoniandre/vue-cal-v4#122]
Hi,
I like vue-cal very much as simple to use. I just wanted to make out of business hours in different color in calendar and scroll position to business hours by default. Any solution to get this result? at least showing background color (for example office open at 9am , so I need to show before 9 am in grey or other color in weeek, and day view).

**Comment by antoniandre** [comment]
Hi @nizarbathery, 
Thanks for your feedback.
Currently there is no direct way to do this but I will add such an option in a future release.

For now, what you could do, is put a grey background to vue-cal then add 1 background event (`background: true`) on each working day from 9am to end of office hours with a 'working-hours' css class that has a white background.

I know it's not perfect but hope it helps a little bit.
Otherwise the easiest is use `time-from` and `time-to` as delimiters of the company working hours.

**Comment by nizarbathery** [comment]
Hi @antoniandre, 
Thank you for the quick response, I hope your suggestion will work well. I've been waiting for your next release to get the single click for adding new event on clicked cell. I am not sure if I add a background event will conflict or work well to add new event in business hours (background event). I need to add new appointment when user single click on empty cell and edit when user single click on existing event. I believe this can be done in coming release. Any news on new version release date?

**Comment by kstankov** [comment]
Hi.
First, thanks for this component, Antoni. Great work!
I also want to ask, if it is possible to scroll the calendar (Week, Day view) to current time.
Thanks in advance!

**Comment by antoniandre** [comment]
Hi @kstankov, 
Sorry I missed your question, thanks for your feedback.
There is no built-in support to do that may come later.
But here is a simple way to do a scroll: 

```js
document.getElementsByClassName('vuecal__bg')[0].scrollTo({ top: 300, behavior: 'smooth' })
```

you can put the y position you want, and remove the smooth param if you want.

**Comment by kurorido** [comment]
In week view, I use `MutationObserver` to scroll to current time

In mounted
```
    this.observer = new MutationObserver(mutations => {
      mutations.forEach(mutation => {
        if (mutation.target.className === 'vuecal__now-line') {
          this.$el.querySelector('.vuecal__bg').scrollTo({
            top: parseInt(mutation.target.style.top)
          })
          // disconnect after scrolling is done
          this.observer.disconnect()
        }
      })
    })
    this.observer.observe(this.$el, {
      attributes: true,
      subtree: true
    })
```

also disconnect in destroy

```
    if (this.observer) {
      this.observer.disconnect()
    }
```

**Comment by antoniandre** [comment]
Hi @nizarbathery, @kstankov, @kurorido,

This feature (business hours background) in now available in version 2.23.0.
https://antoniandre.github.io/vue-cal/#ex--special-hours

I am sorry it took so long to pick this feature request and in the end it was pretty fast to develop. 😄 
Hope it helps, still!

**Comment by antoniandre** [comment]
The feature is out so I'm closing the issue.
Please reopen if you find a problem.
